piptools/sync.py

```diff
@@ -59,7 +60,7 @@ def dependency_tree(
 
```

Comment:
Do we have to worry about `root_key` and the `installed_keys` keys potentially being differently formatted at the time of  the `if root_key in installed_keys` check?

Reply:
Valid point. Thanks! This is why we should use `NormalizedName` type instead of str for package names. I'm going to address that in a follow-up PR as commented [here](https://github.com/jazzband/pip-tools/pull/1919/files#r1265535387).